zospy/zpcore.py

```diff
@@ -391,21 +410,31 @@ def connect_as_extension(self, instancenumber: int = 0) -> bool:
 
         self.Application = self.Connection.ConnectAsExtension(instancenumber)
 
-        if self.Application.IsValidLicenseForAPI:
-            return True
-        else:
+        if not self.Application.IsValidLicenseForAPI:
```

Comment:
A `ConnectionRefusedError` is now also raised if `return_primary_system` is `False`, which introduces a breaking change. This also makes the return statement redundant, since it is unreachable.

```suggestion
        if not self.Application.IsValidLicenseForAPI:
            logger.critical("OpticStudio Licence is not valid for API, connection not established")
            
            if return_primary_system:
                raise ConnectionRefusedError("OpticStudio Licence is not valid for API, connection not established")
            
            return False
        
        if return_primary_system:
            return self.get_primary_system()
            
        return True
```